File: backend/internal/dto/docker.go
package dto

type DockerContainerActionPath struct {
	ID string `uri:"id" binding:"required"`
}

type DockerContainerInfo struct {
	ID     string `json:"id"`
	Name   string `json:"name"`
	Image  string `json:"image"`
	State  string `json:"state"`
	Status string `json:"status"`
}

type ListDockerContainersResult struct {
	Containers []DockerContainerInfo `json:"containers"`
}

type DockerContainerActionResult struct {
	ID    string `json:"id"`
	State string `json:"state"`
}

type DockerImageInfo struct {
	ID       string   `json:"id"`
	RepoTags []string `json:"repo_tags"`
	Size     uint64   `json:"size"`
}

type ListDockerImagesResult struct {
	Images []DockerImageInfo `json:"images"`
}

File: frontend/src/types/docker.ts
export interface DockerContainerInfo {
  id: string;
  name: string;
  image: string;
  state: string;
  status: string;
}

export interface ListDockerContainersResult {
  containers: DockerContainerInfo[];
}

export interface DockerContainerActionResult {
  id: string;
  state: string;
}

export interface DockerImageInfo {
  id: string;
  repo_tags: string[];
  size: number;
}

export interface ListDockerImagesResult {
  images: DockerImageInfo[];
}
